api/src/main/java/com/flipkart/poseidon/api/APIBuildable.java

```diff
@@ -84,7 +84,7 @@ public void build(Request request, Response response, Map<String, Object> model)
     @Override
     public LinkedHashSet<Filter> getFilters(Request request) throws InternalErrorException {
         LinkedHashSet<Filter> filters = new LinkedHashSet<>();
-        filters.add(new ParamValidationFilter(pojo.getParams(), configuration));
+        filters.add(PoseidonLegoSet.wrapFilter(new ParamValidationFilter(pojo.getParams(), configuration)));
```

Comment:
APIBuildable must have legoSet object. You can use that instead of making this static.